tests/TestCase/Listener/SearchListenerTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace Crud\Test\TestCase\Listener;
+
+use Crud\TestSuite\TestCase;
+
+/**
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ */
+class SearchListenerTest extends TestCase
+{
+
+    /**
+     * Test implemented events
+     *
+     * @return void
+     */
+    public function testImplementedEvents()
+    {
+        $listener = $this
+            ->getMockBuilder('\Crud\Listener\SearchListener')
+            ->setMethods(null)
+            ->disableoriginalConstructor()
+            ->getMock();
+
+        $result = $listener->implementedEvents();
+        $expected = [
+            'Crud.beforePaginate' => ['callable' => 'beforePaginate']
+        ];
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * Test beforePaginate
+     *
+     * @return void
+     */
+    public function testBeforePaginate()
+    {
+
```

Comment:
Unsure here:

1. Is it necessary to test because it's more or less `FOC/Search` land.
2. Testable? Because it's only a composer suggestion, it means the only thing I could test is the `RumetimeException`.